### PR TITLE
T21756 autoinstall from usb fixups

### PR DESCRIPTION
**Pull request by pwithnall**
Some fixups for #188 which address all my review comments so far. I’ll comment further on #188 itself.

### CODE REVIEW

File: test-common/utils.c [hunk 1]
```diff
@@ -2932,7 +2932,7 @@ eos_test_client_prepare_volume (EosTestClient *client,
                                                                                "eos-updater-prepare-volume",
                                                                                "eos-updater-prepare-volume",
                                                                                NULL);
-  g_autofree gchar *libeos_updater_util_path = g_test_build_filename (G_TEST_DIST,
+  g_autofree gchar *libeos_updater_util_path = g_test_build_filename (G_TEST_BUILT,
```
Comment (mwleeds) [inline test-common/utils.c:2935]: But that folder is distributed not built? I don't understand this change.

File: eos-updater-prepare-volume/eos-updater-prepare-volume [hunk 2]
```diff
@@ -176,12 +176,26 @@ class VolumePreparer:
 
         return runtimes
 
-    def _add_autoinstall_flatpaks(self, flatpak_collection_refs):
-        applied_actions = EosUpdaterFlatpakInstaller.determine_flatpak_ref_actions_to_check(None)
-        for filename in applied_actions:
-            for action in applied_actions[filename]:
-                if action.type != EosUpdaterUtil.FlatpakRemoteRefActionType.UNINSTALL:
-                    flatpak_collection_refs.append((action.get_collection_id(), action.get_ref()))
+    def _get_autoinstall_flatpaks(self, flatpak_collection_refs):
```
Comment (mwleeds) [inline eos-updater-prepare-volume/eos-updater-prepare-volume]: It seems like there's no longer any reason to have this argument with the way you've rewritten things

File: eos-updater-prepare-volume/eos-updater-prepare-volume [hunk 3]
```diff
@@ -27,10 +27,9 @@ import gi
 gi.require_version('Flatpak', '1.0')
 gi.require_version('GLib', '2.0')
 gi.require_version('OSTree', '1.0')
-gi.require_version('EosUpdaterFlatpakInstaller', '0')
 gi.require_version('EosUpdaterUtil', '0')
 from gi.repository import Flatpak, GLib, OSTree  # noqa
-from gi.repository import EosUpdaterFlatpakInstaller, EosUpdaterUtil
+from gi.repository import EosUpdaterUtil  # noqa
```
Comment (mwleeds) [inline eos-updater-prepare-volume/eos-updater-prepare-volume:32]: What does `# noqa` mean here?

**Review by mwleeds — CHANGES_REQUESTED** [review]

File: test-common/utils.c (hunk 1 above)
Comment (pwithnall) [reply test-common/utils.c:2935]: The path which is produced here is set as the `GI_TYPELIB_PATH` in the environment of `eos-updater-prepare-volume`. The `.typelib` files are built at build time, and are not distributed — so they will be present only in the `G_TEST_BUILT` path for `libeos-updater-util`.

File: eos-updater-prepare-volume/eos-updater-prepare-volume (hunk 3 above)
Comment (pwithnall) [reply eos-updater-prepare-volume/eos-updater-prepare-volume:32]: It stops linting tools like `python3-flake8` emitting a warning on this line. I can’t remember which warning they emit — something about `EosUpdaterUtil` not being a known part of `gi.repository` iirc.

Comment (pwithnall) [reply eos-updater-prepare-volume/eos-updater-prepare-volume:32]: (We run `flake8` as part of `check-local`.)

File: eos-updater-prepare-volume/eos-updater-prepare-volume (hunk 2 above)
Comment (pwithnall) [reply eos-updater-prepare-volume/eos-updater-prepare-volume]: Ah yes. :+1:

Comment (pwithnall) [reply eos-updater-prepare-volume/eos-updater-prepare-volume]: Fixup pushed

**Pull request merged**